src/stores/widgets/SpectralProfileWidget/SpectralProfileWidgetStore.ts

```diff
@@ -316,16 +316,12 @@ export class SpectralProfileWidgetStore extends RegionWidgetStore {
         this.intensityConversion = undefined;
         this.setIntensityUnit(this.effectiveFrame?.unit);
 
-        reaction(
-            () => this.effectiveFrame,
-            frame => {
-                if (frame) {
-                    this.momentRegionId = RegionId.ACTIVE;
-                    this.setIntensityUnit(frame.unit);
-                    this.updateRanges();
-                }
+        autorun(() => {
+            if (this.effectiveFrame) {
+                this.updateRanges();
+                this.momentRegionId = RegionId.ACTIVE;
```

Comment:
may change to `this.selectMomentRegion(RegionId.ACTIVE)` to remove mobx warning
![image](https://user-images.githubusercontent.com/43841102/150739452-5bcc1369-6f2d-47e5-8264-40ac4fb817af.png)

Reply:
revised